### ISSUE TITLE
cf push fails if paths are too long (more than about 260 chars) on windows (ref #716)

### ISSUE THREAD

**Issue by d024441** [cloudfoundry/cli#728]
Hi @krishicks,

this is referring issue #716 (which have been closed already).

Unfortunately, also the actual cf windows client doesn't fix the issue:
```
>cf --version
cf version 6.14.1+dc6adf6-2015-12-22
```
To reproduce the error, please try to push files within a windows path longer than 260 bytes:
```
Using route ...
FAILED
Error processing app files in ...: GetFileAttributesEx <long path>: The system
cannot find the path specified.
```

**Comment by dkoper** [comment]
I can reproduce this by taking the same app from #564, removing its .cfignore and do a `cf push` (cf version 6.14.1+450905f-2016-01-11).
We'll investigate.

```
Using manifest file D:\projects\cf\meanjs\manifest.yml

Updating app mean in org dies-test / space dev as diesk@fast.au.fujitsu.com...
OK

Creating route mean-supergeneric-acetonitrile.a1-app.cf-app.com...
OK

Binding mean-supergeneric-acetonitrile.a1-app.cf-app.com to mean...
OK

FAILED
Error processing app files in 'D:\projects\cf\meanjs': GetFileAttributesEx D:\projects\cf\meanjs\node_modules\grunt-contrib-cssmin\node_modules\maxmin\node_modu
les\pretty-bytes\node_modules\meow\node_modules\normalize-package-data\node_modules\validate-npm-package-license\node_modules\spdx-correct\node_modules\spdx-lic
ense-ids\LICENSE: The system cannot find the path specified.
```

**Comment by krishicks** [comment]
This was fixed only for apps containing a .cfignore file. If your app actually requires apps with paths longer than 260 characters, it will still fail. The fix we made previously was specifically for node.js apps that have enough nested node_modules directories to break `cf push`, and those files aren't actually required to be pushed.

**Comment by dkoper** [comment]
Hi @d024441 

We have made a fix. Can you pushing your app using our edge binary?
https://github.com/cloudfoundry/cli#downloads

I don't get the error you mentioned with the app from the original issue anymore, but I do get an npm error from the buildpack. I don't know if that's related.

**Comment by d024441** [comment]
Hi,
I pushed the app with the 'edge binary', but unfortunately got the same error as before:

```
C:\...\>cf --version
cf version 6.15.0+fa1bfe2-2016-01-12

C:\...\>cf push myapp
Using manifest file C:\...\manifest.yml

Updating app myapp in org ... / space ... as ...
OK

Using route ...
FAILED
Error processing app files in 'C:\...': GetFileAttributesEx C:\<long path name>: The system cannot find the path specified.
```
The following information might be of help:
- Go issue: https://github.com/golang/go/issues/3358
- Windows API (MAX_PATH):
 - https://msdn.microsoft.com/de-de/library/windows/desktop/aa364944%28v=vs.85%29.aspx
 - https://msdn.microsoft.com/de-de/library/windows/desktop/aa365247%28v=vs.85%29.aspx#maxpath  

- https://github.com/d024441/gofileutils/commit/a309e297c171032ee97da76fd472e135ae77abb3

**Comment by dkoper** [comment]
Hi Stefan,

6.15.0+fa1bfe2-2016-01-12 is not an edge binary, it's our latest official release from last week. It doesn't contain the fix. You can get the Edge binary for Windows here: "https://cli.run.pivotal.io/edge?arch=windows64&source=github". The date at the end will be Jan 19th or 20th.

If it doesn't work, we'll need an app with steps how to reproduce your issue locally.

Cheers,
Dies Koper
CF CLI PM

**Comment by d024441** [comment]
Hi Dies,

thanks - sorry for using the wrong binary for re-testing.
cf version 6.15.0+0186957-2016-01-19 works fine in my test case.

Regards,
stefan

**Comment by dkoper** [comment]
FYI, this has been included in cf CLI v6.16.0.
https://github.com/cloudfoundry/cli#downloads

Regards,
Dies Koper
CF CLI PM
